src/main/java/seedu/address/model/appointment/FilterAppointmentPredicate.java

```diff
@@ -6,22 +6,25 @@
  * Tests that a {@code Appointment}'s {@code startDateTime} and {@code endDateTime} matches the date times given.
  */
 public class FilterAppointmentPredicate implements Predicate<Appointment> {
-    private final StartDateTime startDateTime;
-    private final EndDateTime endDateTime;
+    private final StartDateTime lowerBoundDateTime;
+    private final EndDateTime upperBoundDateTime;
 
     /**
      * Finds appointment based on start and end date times.
-     * @param startDateTime target start date time.
-     * @param endDateTime target end date time.
+     * @param lowerBoundDateTime target start date time.
+     * @param upperBoundDateTime target end date time.
      */
-    public FilterAppointmentPredicate(StartDateTime startDateTime, EndDateTime endDateTime) {
-        this.startDateTime = startDateTime;
-        this.endDateTime = endDateTime;
+    public FilterAppointmentPredicate(StartDateTime lowerBoundDateTime, EndDateTime upperBoundDateTime) {
+        this.lowerBoundDateTime = lowerBoundDateTime;
+        this.upperBoundDateTime = upperBoundDateTime;
     }
 
     @Override
     public boolean test(Appointment appointment) {
-        return startDateTime.compareTo(appointment.getStartDateTime()) <= 0
-                && endDateTime.compareTo(appointment.getEndDateTime()) >= 0;
+        // Notice the ! before the whole statement
+        return !((lowerBoundDateTime.isAfter(appointment.getStartDateTime())
+                && lowerBoundDateTime.isAfter(appointment.getEndDateTime()))
+                || (upperBoundDateTime.isBefore(appointment.getStartDateTime())
+                && upperBoundDateTime.isBefore(appointment.getEndDateTime())));
```

Comment:
Could you break down this entire expression into smaller boolean expressions, each as their own variable?

Reply:
Also, if there is no good reason for adding the ! at the start, please avoid adding such a thing, because it makes understanding the expression even more difficult

E.g. Have you ever not not have had difficulty understanding statements like these?

Reply:
Also, please keep in mind of the 3 cases where we want to filter for intervals
1. Lower bound contained within the appointment date range
2. Upper bound contained within the appointment date range
3. Lower and upper bounds contains the appointment date range (range is larger than the appointment date range)

Reply:
> Also, please keep in mind of the 3 cases where we want to filter for intervals
> 
>     1. Lower bound contained within the appointment date range
> 
>     2. Upper bound contained within the appointment date range
> 
>     3. Lower and upper bounds contains the appointment date range (range is larger than the appointment date range)

Well the case here is just combine 2 case needed to exclude then inverse it boolean wise

Reply:
I understand it's shorter, but right now it sacrifices readability, which is not recommended in this course